compiler/hash-types/src/new/trts.rs

```diff
@@ -0,0 +1,43 @@
+//! Definitions related to traits.
+
+use hash_source::identifier::Identifier;
+use hash_utils::{
+    new_sequence_store_key, new_store_key,
+    store::{DefaultSequenceStore, DefaultStore},
+};
+
+use crate::new::{
+    defs::{DefArgsId, DefMember, DefParamsId},
+    symbols::Symbol,
+};
+
+new_sequence_store_key!(pub TrtMembersId);
+pub type TrtMembersStore = DefaultSequenceStore<TrtMembersId, DefMember<TrtDefId>>;
+pub type TrtMemberId = (TrtMembersId, usize);
+
+/// A trait definition.
+///
+/// Includes a name, a set of parameters for the trait, as well as a set of
+/// members, as well as the name of "Self".
+#[derive(Debug, Clone, Copy)]
+pub struct TrtDef {
+    pub name: Identifier,
+    pub params: DefParamsId,
+    pub members: TrtMembersId,
+
+    /// The name of the "Self" type in the scope of the trait definition.
```

Comment:
This comment is confusing, if this is the name of the type, would it not then use `TermId` instead of just a name?